pkg/proxy/proxy.go

```diff
@@ -177,16 +179,32 @@ func (p *proxy) invokeGrpc(writer http.ResponseWriter, request *http.Request) {
 	}
 }
 
-func writeError(writer http.ResponseWriter, err error) {
+type invocationError struct {
+	Error string `json:"error"`
+}
+
+func writeError(writer http.ResponseWriter, err error, accept string) {
+	writeJSON := accept == "application/json"
```

Comment:
it's rare that the accept header will be literally "application/json". It would be better to find the best match for the accepted mime. We can do this in a follow up.